include/rwqueue.h

```diff
@@ -39,14 +39,27 @@
 #include <atomic>
 #include <condition_variable>
 #include <deque>
+
+#ifdef HAVE_MEMORY_RESOURCE
+	#include <memory_resource>
+#endif
+
 #include <mutex>
 #include <optional>
 #include <vector>
 
 template <typename T>
 class RWQueue {
 private:
-	std::deque<T> queue{}; // faster than: vector, queue, and list
+#ifdef HAVE_MEMORY_RESOURCE
+	std::pmr::unsynchronized_pool_resource pool = {};
```

Comment:
We'll trust you on this 😅 I don't think any of us are familiar with this, we're not exactly C++ gurus 😆

Reply:
Good news is that the rwqueue_tests.cpp appear to be very thorough and exercise the pooled deque as the rwqueue's internal container. (No need for trusted gurus here :genie_man:!) :laughing: